lib/grammers-client/src/types/inline_query.rs

```diff
@@ -0,0 +1,196 @@
+use super::{Chat, ChatMap};
+use crate::{client::Client, utils::generate_random_id, InputMessage};
+use grammers_mtsender::InvocationError;
+use grammers_tl_types as tl;
+use std::sync::Arc;
+
+/// Represents an inline query update, which occurs when you sign in as a bot and a user sends an
+/// inline query such as `@bot query`.
+pub struct InlineQuery {
+    query: tl::types::UpdateBotInlineQuery,
+    client: Client,
+    chats: Arc<ChatMap>,
+}
+
+/// An inline query answer builder.
+pub struct Answer {
+    request: tl::functions::messages::SetInlineBotResults,
+    client: Client,
+}
+
+/// An inline query result.
+///
+/// grammers currently supports results of the following types that implement [`Into<InlineResult>`]:
+/// - [`Article`]
+pub struct InlineResult(tl::enums::InputBotInlineResult);
+
+impl From<InlineResult> for tl::enums::InputBotInlineResult {
+    fn from(result: InlineResult) -> Self {
+        result.0
+    }
+}
+
+impl InlineQuery {
+    pub(crate) fn new(
+        client: &Client,
+        query: tl::types::UpdateBotInlineQuery,
+        chats: &Arc<ChatMap>,
+    ) -> Self {
+        Self {
+            query,
+            client: client.clone(),
+            chats: chats.clone(),
+        }
+    }
+
+    // User that sent the query.
+    pub fn chat(&self) -> &Chat {
```

Comment:
I do wonder if we should return the friendly `User` from methods like these. `chat` is definitely confusing though, because it's not the chat where the inline query was executed. `sender` is consistent with `Message` and now also `CallbackQuery`.

```suggestion
    pub fn sender(&self) -> &Chat {
```

Reply:
It would probably make more sense, since the sender is always a user.